src/main/java/frc/robot/Constants.java

```diff
@@ -1,10 +1,85 @@
+        public static final double[] rl_drivePID = new double[]{0, 0, 0, 0}; // rear left PID controller for drive.
+        public static final double[] rr_drivePID = new double[]{0, 0, 0, 0}; // rear right PID controller for drive.
+
+        public static final double[][] drivePID = new double[][]{ // all the PID configurations for drive.
+                fl_drivePID, fr_drivePID, rl_drivePID, rr_drivePID
+        };
+
+        public static final double TICKS_IN_ENCODER_ANGLE = 1024;
+        public static final double TICKS_IN_ENCODER_DRIVE = 1024;
+        public static final double TICKS_PER_DEGREE_ANGLE = SwerveModule.GEAR_RATIO_ANGLE * TICKS_IN_ENCODER_ANGLE / 360; // ticks per degree. [ticks/deg]
+        public static final double TICKS_PER_METER_DRIVE = SwerveModule.GEAR_RATIO_DRIVE * TICKS_IN_ENCODER_DRIVE / (2 * Math.PI * WHEEL_RADIUS); // ticks per meter. [ticks/m]
+
+        public static final double Rx = 0.75 / 2; // x position of the modules. [m]
+        public static final double Ry = 0.75 / 2; // y position of the modules. [m]
+
+        public static final double MAX_VELOCITY = 4.5; // maximum velocity of the modules. [m/s]
+        public static final double MAX_ROTATIONAL_VELOCITY = 4.5; // maximum rotational velocity of the robot. [rad/s]
+
+        public static final double DEADBAND = 0.02; // deadband for the joysticks.
+
+        public static final int[] fl_ZERO_POSITIONS = new int[]{0, 0};
+        public static final int[] fr_ZERO_POSITIONS = new int[]{0, 0};
+        public static final int[] rl_ZERO_POSITIONS = new int[]{0, 0};
+        public static final int[] rr_ZERO_POSITIONS = new int[]{0, 0};
+
+        public static final int[][] ZERO_POSITIONS = new int[][]{
+                fl_ZERO_POSITIONS, fr_ZERO_POSITIONS, rl_ZERO_POSITIONS, rr_ZERO_POSITIONS // fl, fr, rl, rr. [ticks]
+        };
+    }
+
+    public static class SwerveModule {
+        public static final double GEAR_RATIO_ANGLE = 1;
+        public static final double GEAR_RATIO_DRIVE = 1;
+        public static final double Kv = 1;
+        public static final double Ka = 1;
+        public static final Matrix<N1, N1> A = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(
+                -Kv / Ka
+        );
+        public static final Matrix<N1, N1> B = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(
+                1 / Ka
+        );
+        public static final Matrix<N1, N1> C = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(
+                1
+        );
+        public static final Matrix<N1, N1> D = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(
+                0
+        );
+
+        public static final Matrix<N1, N1> MODEL_TOLERANCE = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(
+                0.8
+        );
+        public static final Matrix<N1, N1> SENSOR_TOLERANCE = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(
+                0.2
+        );
```

Comment:
Why all this excessive whitespace?

Reply:
Why not? It looks better.

Reply:
You usually split function calls when you have many arguments _and_ they are long. Here you have a single argument _and_ it's only three characters long. Long code forces readers to scroll more and think more. Consider this instead, much shorter and clearer: 
```suggestion
        public static final Matrix<N1, N1> A = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(-Kv / Ka);
        public static final Matrix<N1, N1> B = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(1 / Ka);
        public static final Matrix<N1, N1> C = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(1);
        public static final Matrix<N1, N1> D = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(0);

        public static final Matrix<N1, N1> MODEL_TOLERANCE = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(0.8);
        public static final Matrix<N1, N1> SENSOR_TOLERANCE = new MatBuilder<>(Nat.N1(), Nat.N1()).fill(0.2);
```